src/NPacMan.Game/Game.cs

```diff
@@ -221,26 +221,26 @@ async Task IGameActions.MovePacMan(BehaviorContext<GameState, Tick> context, Gam
                 PacMan = newPacMan;
             }
 
-            if (TryHasDied(out var ghost))
+            if (TryHasCollidedWithGhost(out var ghost))
             {
                 await context.Raise(gameStateMachine.GhostCollision, new GhostCollision(ghost!));
             }
 
             if (Coins.Contains(newPacMan.Location))
             {
-                await context.Raise(gameStateMachine.CoinEaten, new CoinEaten(newPacMan.Location));
+                await context.Raise(gameStateMachine.CoinCollision, new CoinCollision(newPacMan.Location));
             }
 
             if (PowerPills.Contains(newPacMan.Location))
             {
-                await context.Raise(gameStateMachine.PowerPillEaten, new PowerPillEaten(newPacMan.Location));
+                await context.Raise(gameStateMachine.PowerPillCollision, new PowerPillCollision(newPacMan.Location));
             }
         }
 
         private async Task MoveGhosts(BehaviorContext<GameState, Tick> context, GameStateMachine gameStateMachine)
         {
             ApplyToGhosts(ghost => ghost.Move(this));
-            if (TryHasDied(out var ghost))
+            if (TryHasCollidedWithGhost(out var ghost))
```

Comment:
If 2 ghosts are edible and we hit 2 at the same time then we'd walk through one of them while eating the other?

Reply:
Also the returned list needs to be sorted so that non-edible ghosts are returned first.